physical/raft/fsm.go

```diff
@@ -68,6 +78,22 @@ func (f *FSMEntry) String() string {
 	return fmt.Sprintf("Key: %s. Value: %s", f.Key, hex.EncodeToString(f.Value))
 }
 
+func (f *FSMEntry) IsTxError() bool {
```

Comment:
@cipherboy : I'm not sure about the pattern here in Go, but are you sure there is no need to check if "f" is a  nil pointer or not before using it in the function ("class method" )  ?

Reply:
Technically, yes: https://go.dev/play/p/sFM5BnM8vDz 

However, checking for a nil value in a method call isn't common practice in Go: as far as I can tell with semgrep, this is never done across the entire code base:

```yml
rules:
  - id: pointer-checked-nil
    patterns:
      - pattern-inside: |
          func ($FOO *$TYPE) $FUNC(...) {
             ...
          }
      - pattern:
          if $FOO == nil { ... }
    message: "Checking pointer receiver against nil"
    languages: [go]
    severity: ERROR
```

I have not seen it done either in other code bases. TBH, I'm not sure how you'd handle this, return an error? Now every method needs to return an error where one didn't exist, rather than panicking. 

In our case, we know from construction that this will never be nil.